mocha-appveyor-reporter.js

```diff
@@ -0,0 +1,131 @@
+var util = require('util'),
+    Base = require('mocha').reporters.Base;
+
+function AppVeyorReporter(runner) {
+  Base.call(this, runner);
+  var request = require('request-json'),
+    deasync = require('deasync'),
+    endpoint = process.env.APPVEYOR_API_URL,
+    log = console.log.bind(console),
+    error = console.error.bind(console),
+    warn = console.warn.bind(console),
+    logEntries = [],
+    client = endpoint ? request.createClient(endpoint) : null;
+
+  if(!endpoint) {
+    warn("APPVEYOR_API_URL environment variable is not set, will not report to appveyor.");
+  }
+
+  /**
+   * Takes a test result and maps it to the output format that AppVeyor accepts.
+   * @param test Result generated by mocha.
+   * @returns {{testName: string, testFramework: string, fileName: string, outcome: string, durationMilliseconds: Number, StdOut: string, StdErr: string}}
+   */
+  var mapTest = function(test) {
+    var stdout = "";
+    var stderr = "";
+    logEntries.forEach(function (data) {
+      var msg = util.format.apply(util, data.msg);
+      if (data.type === 'out') {
+        stdout = stdout + msg + '\n';
+      } else {
+        stderr = stderr + msg + '\n';
+      }
+    });
+    return {
+      testName: test.fullTitle(),
+      testFramework: 'Mocha',
+      fileName: test.file,
+      outcome: test.state === "passed" ? "Passed" : null,
```

Comment:
It was not intentional, but that is a good point, while the outcome will eventually be replaced (if Ignored or Failed) I should clean that up.  Same with your point above about initializing the client.